### PR TITLE
add the sparsehash recipe

### PR DESCRIPTION
**Pull request by vgauthier**
Do the pull request of the sparsehash recipe from a branch

### CODE REVIEW

**Comment by jakirkham** [comment]
Weird. Still no dice. Well let's see if later commits are better.

File: recipes/sparsehash/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    fn: sparsehash-{{ version }}.tar.gz
+    url: https://github.com/sparsehash/sparsehash/archive/sparsehash-{{ version }}.tar.gz
+    {{ checktype }}: {{ checksum }}
+
+build:
+    number: 0
+    skip: true  # [win]
+
+requirements:
+    build:
```
Comment (jakirkham) [inline recipes/sparsehash/meta.yaml]: Please add `toolchain` to the build dependencies.

File: recipes/sparsehash/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        - pkg-config # [not win]
```
Comment (jakirkham) [inline recipes/sparsehash/meta.yaml]: Please drop the selector as Windows is skipped.

File: recipes/sparsehash/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    run:
+        - pkg-config # [not win]
+
+test:
+    commands:
+        - test -f ${PREFIX}/lib/pkgconfig/libsparsehash.pc # [not win]
+
+about:
+    home: https://github.com/sparsehash/sparsehash
+    license: BSD
```
Comment (jakirkham) [inline recipes/sparsehash/meta.yaml]: Please specify clauses.

File: recipes/sparsehash/build.sh [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    export CC=clang
+    export CXX=clang++
+    export MACOSX_VERSION_MIN="10.7"
+    export MACOSX_DEPLOYMENT_TARGET="${MACOSX_VERSION_MIN}"
+    export CXXFLAGS="${CXXFLAGS} -mmacosx-version-min=${MACOSX_VERSION_MIN}"
+    export CXXFLAGS="${CXXFLAGS} -stdlib=libc++"
+    export LDFLAGS="${LDFLAGS} -mmacosx-version-min=${MACOSX_VERSION_MIN}"
+    export LDFLAGS="${LDFLAGS} -stdlib=libc++ -lc++"
+    export LINKFLAGS="${LDFLAGS}"
+fi
```
Comment (jakirkham) [inline recipes/sparsehash/build.sh]: Please drop the above `if` statement and replace with this.

```
# FIXME: This is a hack to make sure the environment is activated.
# The reason this is required is due to the conda-build issue
# mentioned below.
#
# https://github.com/conda/conda-build/issues/910
#
source activate "${CONDA_DEFAULT_ENV}"
```

File: recipes/sparsehash/build.sh [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+# Setup the boost building, this is fairly simple.
+./configure --prefix=${PREFIX}
+make
```
Comment (jakirkham) [inline recipes/sparsehash/build.sh]: Please add `make check` or similar.

File: recipes/sparsehash/build.sh [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+make install
+
+exit 0
```
Comment (jakirkham) [inline recipes/sparsehash/build.sh]: Please drop as it is not needed.

File: recipes/sparsehash/build.sh [hunk 7]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+set -e
+
+UNAME="$(uname)"
+export CFLAGS="-O3"
+export CXXFLAGS="-O3"
```
Comment (jakirkham) [inline recipes/sparsehash/build.sh]: Please move after the `source activate` section proposed below.

File: recipes/sparsehash/build.sh [hunk 8]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+set -e
+
+UNAME="$(uname)"
```
Comment (jakirkham) [inline recipes/sparsehash/build.sh]: This can be dropped.

File: recipes/sparsehash/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+requirements:
+    build:
+        - pkg-config # [not win]
+    run:
+        - pkg-config # [not win]
+
+test:
+    commands:
+        - test -f ${PREFIX}/lib/pkgconfig/libsparsehash.pc # [not win]
```
Comment (jakirkham) [inline recipes/sparsehash/meta.yaml]: Are there any libraries or includes we can check for? Please add them here too. Also please drop the selector as it is unneeded.

**Comment by vgauthier** [comment]
@jakirkham I tried to address all of your comments

File: recipes/sparsehash/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+        - toolchain
+        - pkg-config
+
+test:
+    commands:
+        - test -f ${PREFIX}/lib/pkgconfig/libsparsehash.pc
+
+about:
+    home: https://github.com/sparsehash/sparsehash
+    license: BSD
```
Comment (jakirkham) [inline recipes/sparsehash/meta.yaml]: Please specify clauses.

File: recipes/sparsehash/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+    build:
+        - toolchain
+        - pkg-config
+    run:
+        - toolchain
+        - pkg-config
+
+test:
+    commands:
+        - test -f ${PREFIX}/lib/pkgconfig/libsparsehash.pc
```
Comment (jakirkham) [inline recipes/sparsehash/meta.yaml]: Please add file tests for other libraries.

File: recipes/sparsehash/meta.yaml [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+    number: 0
+    skip: true  # [win]
+    
+requirements:
+    build:
+        - toolchain
+        - pkg-config
+    run:
+        - toolchain
+        - pkg-config
```
Comment (jakirkham) [inline recipes/sparsehash/meta.yaml]: Neither of these should be needed as run dependencies.

File: recipes/sparsehash/build.sh [hunk 13]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+source activate "${CONDA_DEFAULT_ENV}"
+
+# Setup the boost building, this is fairly simple.
+export CFLAGS="-O3"
```
Comment (jakirkham) [inline recipes/sparsehash/build.sh]: Change to `export CFLAGS="${CFLAGS} -O3"

File: recipes/sparsehash/build.sh [hunk 14]
```diff
@@ -0,0 +1,12 @@
...
+export CXXFLAGS="-O3"
```
Comment (jakirkham) [inline recipes/sparsehash/build.sh]: Change to `export CXXFLAGS="${CXXFLAGS} -O3"`.

**Comment by jakirkham** [comment]
Looks pretty good. Couple more things to change.

File: recipes/sparsehash/meta.yaml [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+    commands:
+        - test -f "${PREFIX}/lib/pkgconfig/libsparsehash.pc"  # [unix]
+        - test -r "${PREFIX}/include/sparsehash"  # [unix]
+        - test -r "${PREFIX}/include/sparsehash/internal"  # [unix]
+        - test -r "${PREFIX}/include/google/"  # [unix]
+        - test -r "${PREFIX}/include/google/sparsehash"  # [unix]
+
+about:
+    home: https://github.com/sparsehash/sparsehash
+    license: BSD
```
Comment (jakirkham) [inline recipes/sparsehash/meta.yaml]: Looks like this should be `BSD 3-Clause`. Please update accordingly.

File: recipes/sparsehash/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipes/sparsehash/meta.yaml]: Done below.

**Comment by jakirkham** [comment]
I still don't know why CircleCI didn't run here, but was able to run it locally and determined it would pass.

**Comment by jakirkham** [comment]
LGTM

File: recipes/sparsehash/build.sh [hunk 16]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+source activate "${CONDA_DEFAULT_ENV}"
```
Comment (jakirkham) [inline recipes/sparsehash/build.sh]: Please add the following comment above this line.

```
# FIXME: This is a hack to make sure the environment is activated.
# The reason this is required is due to the conda-build issue
# mentioned below.
#
# https://github.com/conda/conda-build/issues/910
#
```

**Comment by jakirkham** [comment]
Thanks @vgauthier.

**Comment by jakirkham** [comment]
@donovanhide @Dekken, just letting you know we have added `sparsehash` to [conda-forge]( http://conda-forge.github.io/ ). This will allow users to easily install this with [`conda`]( http://conda.pydata.org/docs/ ) (`conda install -c conda-forge sparsehash`). The maintenance burden here is light as all testing releases are built and deployed automatically with CI. Changes to how it is being built can be controlled by changing the recipe in [this repo]( https://github.com/conda-forge/sparsehash-feedstock ). If you have any interest in being added as a maintainer (to make new releases for instance), we would love to have you on board. Thanks for your work on `sparsehash`. 😄

**Pull request merged**